tox.ini

```diff
@@ -10,6 +11,11 @@ deps =
 passenv =
     KUBECONFIG
 
+[testenv:unit]
+deps =
+    {[testenv]deps}
+commands = pytest -v --cov=. --cov-report=term-missing {toxinidir}/tests/unit
```

Comment:
The `--cov=. --cov-report=term-missing` bits seem like something we'd want to store as an artifact in Github Actions, maybe with `--cov-report html`. If you're up for it, it should be as simple as this:

```yaml
    - name: Upload coverage artifacts
      if: failure()
      uses: actions/upload-artifact@v2
      with:
        name: coverage
        path: <COV-FILE-NAME>
```

Otherwise we can kick that can down the road

Reply:
@knkski Yes, agree that would be a nice addition, I really like the coverage report :ok_hand: . I'll add that feature in another PR as this one is for adding unit testing exclusively.